pkg/ibmcsidriver/controller_helper_test.go

```diff
@@ -301,6 +300,72 @@ func TestGetVolumeParameters(t *testing.T) {
 			expectedStatus: true,
 			expectedError:  fmt.Errorf("volume capabilities are empty"),
 		},
+		{
+			testCaseName: "Region present but zone not given as parameter",
+			request: &csi.CreateVolumeRequest{Name: volumeName, CapacityRange: &csi.CapacityRange{RequiredBytes: 11811160064, LimitBytes: utils.MinimumVolumeSizeInBytes + utils.MinimumVolumeSizeInBytes},
+				VolumeCapabilities: []*csi.VolumeCapability{{AccessMode: &csi.VolumeCapability_AccessMode{Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER}}},
+				Parameters: map[string]string{Profile: "general-purpose",
+					Region:        "us-south-test",
+					Tag:           "test-tag",
+					ResourceGroup: "myresourcegroups",
+					Encrypted:     "false",
+					EncryptionKey: "key",
+					ClassVersion:  "",
+					Generation:    "generation",
+					IOPS:          noIops,
+				},
+			},
+			expectedVolume: &provider.Volume{Name: &volumeName,
+				Capacity: &volumeSize,
+				VPCVolume: provider.VPCVolume{VPCBlockVolume: provider.VPCBlockVolume{
+					Tags: []string{createdByIBM},
+				},
+					Profile:       &provider.Profile{Name: "general-purpose"},
+					ResourceGroup: &provider.ResourceGroup{ID: "myresourcegroups"},
+				},
+				Region: "us-south-test",
+				Iops:   &noIops,
+				Az:     "testzone",
+			},
+			expectedStatus: true,
+			expectedError:  fmt.Errorf("zone parameter is empty in storage class for region us-south-test"),
+		},
+		{
+			testCaseName: "Region and Zone not given as parameter from SC",
+			request: &csi.CreateVolumeRequest{Name: volumeName, CapacityRange: &csi.CapacityRange{RequiredBytes: 11811160064, LimitBytes: utils.MinimumVolumeSizeInBytes + utils.MinimumVolumeSizeInBytes},
+				VolumeCapabilities: []*csi.VolumeCapability{{AccessMode: &csi.VolumeCapability_AccessMode{Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER}}},
+				Parameters: map[string]string{Profile: "general-purpose",
+					Tag:           "test-tag",
+					ResourceGroup: "myresourcegroups",
+					Encrypted:     "false",
+					EncryptionKey: "key",
+					ClassVersion:  "",
+					Generation:    "generation",
+					IOPS:          noIops,
+				},
+				AccessibilityRequirements: &csi.TopologyRequirement{Preferred: []*csi.Topology{{Segments: map[string]string{
+					utils.NodeRegionLabel: "myregion",
+					utils.NodeZoneLabel:   "myzone",
+				},
+				},
+				},
+				},
+			},
+			expectedVolume: &provider.Volume{Name: &volumeName,
+				Capacity: &volumeSize,
+				VPCVolume: provider.VPCVolume{VPCBlockVolume: provider.VPCBlockVolume{
+					Tags: []string{createdByIBM},
+				},
+					Profile:       &provider.Profile{Name: "general-purpose"},
+					ResourceGroup: &provider.ResourceGroup{ID: "myresourcegroups"},
+				},
+				Region: "myregion",
+				Iops:   &noIops,
+				Az:     "myzone",
+			},
+			expectedStatus: true,
+			expectedError:  nil,
+		},
```

Comment:
Can we add one more test case like `zone is given but region is empty`

Reply:
I think is it is already there 
https://github.com/kubernetes-sigs/ibm-vpc-block-csi-driver/pull/67/files#:~:text=name%3A%20%22Zone-,provided,-but%20region%20not